lib/Default/SmrPlayer.class.inc

```diff
@@ -1725,37 +1728,46 @@ class SmrPlayer extends AbstractSmrPlayer {
 
 		$message = word_filter($message);
 
-		switch ($messageTypeID) {
-			case MSG_GLOBAL: //We don't send globals to the box here or it gets done loads of times.
-				$expires = 3600; // 1h
-			break;
-			case MSG_PLAYER:
-				$expires = 86400 * 31;
-			break;
-			case MSG_PLANET:
-				$expires = 86400 * 7;
-			break;
-			case MSG_SCOUT:
-				$expires = 86400 * 3;
-			break;
-			case MSG_POLITICAL:
-				$expires = 86400 * 31;
-			break;
-			case MSG_ALLIANCE:
-				$expires = 86400 * 31;
-			break;
-			case MSG_ADMIN:
-				$expires = 86400 * 365;
-			break;
-			case MSG_CASINO:
-				$expires = 86400 * 31;
-			break;
-			default:
-				$expires = 86400 * 7;
+		// If expires not specified, use default based on message type
+		if ($expires === false) {
+			switch ($messageTypeID) {
+				case MSG_GLOBAL: //We don't send globals to the box here or it gets done loads of times.
+					$expires = 3600; // 1h
+				break;
+				case MSG_PLAYER:
+					$expires = 86400 * 31;
+				break;
+				case MSG_PLANET:
+					$expires = 86400 * 7;
+				break;
+				case MSG_SCOUT:
+					$expires = 86400 * 3;
+				break;
+				case MSG_POLITICAL:
+					$expires = 86400 * 31;
+				break;
+				case MSG_ALLIANCE:
+					$expires = 86400 * 31;
+				break;
+				case MSG_ADMIN:
+					$expires = 86400 * 365;
+				break;
+				case MSG_CASINO:
+					$expires = 86400 * 31;
+				break;
+				default:
+					$expires = 86400 * 7;
+			}
+			$expires += TIME;
```

Comment:
It might be nice to have the `$expires` that is passed in just be how long before expiration, eg "3 days", rather than a fixed time as it is currently ("2018-04-29 17:40:00")

Reply:
I agree that would be nice, but doing it for this function only would be a bit inconsistent since there are a lot of other functions that have `$expires` options that are fixed times. I'd want to change it globally if at all.